lib/screens/create_region_screens/interactor/usecases/validate_region_id_usecase.dart

```diff
@@ -0,0 +1,12 @@
+import 'package:seeds/datasource/remote/api/region_repository.dart';
+import 'package:seeds/datasource/remote/model/region_model.dart';
+import 'package:seeds/domain-shared/base_use_case.dart';
+
+class ValidateRegionIdUseCase extends InputUseCase<RegionModel, String> {
+  final RegionRepository _regionRepository = RegionRepository();
+
+  @override
+  Future<Result<RegionModel>> run(String input) async {
+    return _regionRepository.getRegionById(input);
```

Comment:
i think this function that you created can return a null?

Reply:
shold  this be Future<Result<RegionModel?>>

Reply:
it can, fixed